libs/components/core/src/lib/modules/id/id.directive.spec.ts

```diff
@@ -32,15 +40,15 @@ describe('ID directive', () => {
     await expectAsync(fixture.nativeElement).toBeAccessible();
   });
 
-  it('should include a timestamp in the generated ID to avoid clashes across sessions', () => {
-    spyOn(Date.prototype, 'getTime').and.returnValue(12345);
+  it('should call the generateId method from the ID service', () => {
+    const idSvc = TestBed.inject(SkyIdService);
+
+    const spy = spyOn(idSvc, 'generateId');
 
     const fixture = TestBed.createComponent(IdDirectiveFixtureComponent);
 
     fixture.detectChanges();
 
-    const input1El = fixture.debugElement.query(By.css('.test-input1'));
-
-    expect(input1El.attributes['id']).toMatch(/sky-id-gen__12345__[0-9]+/);
+    expect(spy).toHaveBeenCalledTimes(2);
```

Comment:
I think you want to do a bit more than just verifying the function was called; I'd add make sure the value returned by the function was used to set the element's ID. If you provide `SkyIdService` and use `useValue` then you can have a `generateId()` function that returns a constant value which you can check in this test. Then you don't need the spy.

```
TestBed.configureTestingModule({
  providers: [
    {
      provide: SkyIdService,
      useValue: {
        generateId: () => '12345'
      }
    }
  ]
});
```

```
    const fixture = TestBed.createComponent(IdDirectiveFixtureComponent);

    fixture.detectChanges();

    const input1El = fixture.debugElement.query(By.css('.test-input1'));

    expect(input1El.attributes['id']).toBe('12345');
```